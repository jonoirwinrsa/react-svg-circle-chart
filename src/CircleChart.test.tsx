import { render } from '@testing-library/react';
import * as React from 'react';
import CircleChart from '../build';

const values = [
  {angle: 45, color: '#47B881'},
  {angle: 50, color: '#EC4C47'},
  {angle: 73, color: '#E2E449'}
];

describe('CircleChart', () => {
  const setup = () => {
    return render(
      <CircleChart values={values} />
    );
  };

  it('should render without crashing', () => {
    setup();
  });

  it('[snapshot] is loaded with default props', () => {
    const {container} = setup();
    expect(container.firstChild).toMatchSnapshot();
  });
});
