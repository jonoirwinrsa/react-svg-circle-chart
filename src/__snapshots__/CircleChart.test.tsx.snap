// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleChart [snapshot] is loaded with default props 1`] = `
<div
  style="position: absolute;"
>
  <svg
    stroke="#E2E449"
    stroke-dasharray="458.6725274241098 999"
    style="overflow: visible; position: absolute; stroke-linecap: round; fill: none; stroke-width: 30; transform: rotate(-90deg); transition: all 1s ease-in-out; margin-top: -200px; height: 200px; width: 200px;"
  >
    <circle
      cx="0"
      cy="0"
      r="100"
    />
  </svg>
  <svg
    stroke="#EC4C47"
    stroke-dasharray="314.1592653589793 999"
    style="overflow: visible; position: absolute; stroke-linecap: round; fill: none; stroke-width: 30; transform: rotate(-90deg); transition: all 1s ease-in-out; margin-top: -200px; height: 200px; width: 200px;"
  >
    <circle
      cx="0"
      cy="0"
      r="100"
    />
  </svg>
  <svg
    stroke="#47B881"
    stroke-dasharray="282.7433388230814 999"
    style="overflow: visible; position: absolute; stroke-linecap: round; fill: none; stroke-width: 30; transform: rotate(-90deg); transition: all 1s ease-in-out; margin-top: -200px; height: 200px; width: 200px;"
  >
    <circle
      cx="0"
      cy="0"
      r="100"
    />
  </svg>
</div>
`;
